scipy/stats/tests/test_multivariate.py

```diff
@@ -164,8 +169,13 @@ def test_mvn_with_covariance(self, size, matrix_type, cov_type_name):
         x1 = mvn.rvs(mean, cov_object, size=size, random_state=rng)
         rng = np.random.default_rng(5292808890472453840)
         x2 = mvn(mean, cov_object, seed=rng).rvs(size=size)
-        assert_close(x1, x)
-        assert_close(x2, x)
+        if isinstance(cov_object, _covariance.CovViaPSD):
+            assert_close(x1, np.squeeze(x))  # for backward compatibility
+            assert_close(x2, np.squeeze(x))
+        else:
+            assert_close(x1.shape, x.shape)
+            assert_close(x2.shape, x.shape)
```

Comment:
```suggestion
            assert_equal(x1.shape, x.shape)
            assert_equal(x2.shape, x.shape)
```

Better to use `assert_equal` since they are all integers and should exactly match.

Reply:
Done in next commit.